solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -294,10 +294,6 @@ public String readConfigName(String collection) throws KeeperException {
     log.debug("Loading collection config from: [{}]", path);
 
     try {
-      if (zkClient.exists(path, true) == false) {
-        log.warn("No collection found at path {}.", path);
-        throw new KeeperException.NoNodeException("No collection found at path: " + path);
-      }
       byte[] data = zkClient.getData(path, null, null, true);
```

Comment:
Small thing, without the check above this will throw a NoNodeException if the path doesn't exist. Maybe this can be wrapped in a try/catch just for the NoNodeException, so that the more user-friendly message used above can be thrown instead:

`throw new KeeperException.NoNodeException("No collection found at path: " + path);`

Just a thought, I'm not even sure it's worth it.

Reply:
As the NoNodeException was still thrown, and the message, even not as precise, is still pretty clear, I left it that way. But if you believe it can be useful, I can add that check with the proper message.

Reply:
I suggest leaving this because it'll be quite moot once the configSet setting moves to state.json.